classes/ActionScheduler_wpPostStore.php

```diff
@@ -466,30 +554,24 @@ public function mark_failure( $action_id ) {
 		}
 	}
 
+	public function get_claim_id( $action_id ) {
+		return $this->get_post_column( $action_id, 'post_password' );
+	}
+
 	public function get_status( $action_id ) {
-		/** @var \wpdb $wpdb */
-		global $wpdb;
-		$sql = $wpdb->prepare( "SELECT post_status FROM {$wpdb->posts} WHERE ID=%d AND post_type=%s", $action_id, self::POST_TYPE );
-		$status = $wpdb->get_var( $sql );
-		switch( $status ) {
-			case 'publish':
-				return self::STATUS_COMPLETE;
-			case 'pending':
-				return self::STATUS_PENDING;
-			case 'trash':
-				return self::STATUS_CANCELED;
-			case self::STATUS_FAILED:
-				return self::STATUS_FAILED;
-			case self::STATUS_RUNNING:
-				return self::STATUS_RUNNING;
-			default:
-				if ( $status === null ) {
-					throw new \InvalidArgumentException( __( 'Invalid action ID. No status found.', 'action-scheduler' ) );
-				} else {
-					throw new \RuntimeException( __( 'Unknown status found for action.', 'action-scheduler' ) );
-				}
+		$status = $this->get_post_column( $action_id, 'post_status' );
+
+		if ( $status === null ) {
+			throw new \InvalidArgumentException( __( 'Invalid action ID. No status found.', 'action-scheduler' ) );
 		}
 
+		return $this->get_action_status_by_post_status( $status );
+	}
+
+	private function get_post_column( $action_id, $column_name ) {
```

Comment:
This should include a docblock.

While this is private and won't be used outside of this class, it may be worth considering whether we want to do any kind of validation on the column name.

Reply:
I'm going to remove this method instead and use `get_post_field()` in its place. That gets us a bit of extra WP magic sauce, even if it will be less efficient than a `$wpdb` query.